src/gui/tray/ActivityItem.qml

```diff
@@ -0,0 +1,264 @@
+                
+                onClicked:  moreActionsButtonContextMenu.popup();
+                
+                Connections {
+                    target: trayWindow
+                    onActiveChanged: {
+                        if (!trayWindow.active) {
+                            moreActionsButtonContextMenu.close();
+                        }
+                    }
+                }
+                
+                Connections {
+                    target: activityListView
+                    
+                    onMovementStarted: {
+                        moreActionsButtonContextMenu.close();
+                    }
+                }
+                
+                Container {
+                    id: moreActionsButtonContextMenuContainer
+                    visible: moreActionsButtonContextMenu.opened
+                    
+                    width: moreActionsButtonContextMenu.width
+                    height: moreActionsButtonContextMenu.height
+                    anchors.right: moreActionsButton.right
+                    anchors.top: moreActionsButton.top
+                    
+                    Menu {
+                        id: moreActionsButtonContextMenu
+                        anchors.centerIn: parent
+                        
+                        // transform model to contain indexed actions with primary action filtered out
+                        function actionListToContextMenuList(actionList) {
+                            // early out with non-altered data
+                            if (activityItem.links.length <= activityListView.maxActionButtons) {
+                                return actionList;
+                            }
+                            
+                            // add index to every action and filter 'primary' action out
+                            var reducedActionList = actionList.reduce(function(reduced, action, index) {
+                                if (!action.primary) {
+                                    var actionWithIndex = { actionIndex: index, label: action.label };
+                                    reduced.push(actionWithIndex);
+                                }
+                                return reduced;
+                            }, []);
+                            
+                            
+                            return reducedActionList;
+                        }
+                        
+                        Repeater {
+                            id: moreActionsButtonContextMenuRepeater
+                            
+                            model: moreActionsButtonContextMenu.actionListToContextMenuList(activityItem.links)
+                            
+                            delegate: MenuItem {
+                                id: moreActionsButtonContextMenuEntry
+                                readonly property int actionIndex: model.modelData.actionIndex
```

Comment:
please remove the 2 properties and just use their value so
```
readonly property int actionIndex: model.modelData.actionIndex
readonly property string label: model.modelData.label
text: label
onTriggered: activityModel.triggerAction(activityItem.itemIndex, actionIndex)
```
becomes
```
text: model.modelData.label
onTriggered: activityModel.triggerAction(activityItem.itemIndex, model.modelData.actionIndex)
```

that is generating some warnings for me (the `int` property in fact)